tests/framework/test_time.py

```diff
@@ -56,94 +62,123 @@ def step_column(sim):
 
-    # Check that the 0th simulant won't step forward
-    sim._population._population.step_size[0] *= 2
-    sim.step()
-    assert len(active_simulants(sim)) == pop_size - 1
-
-    # Show that now that the next_event_time is updated, 0th simulant isn't included in events
-    sim.step()
+    ## Force a next event time update without updating step sizes.
+    ## This ensures (against the current implementation) that we will have a timestep
+    ## that has no simulants aligned. Check that we do the minimum timestep update.
+    sim._population._population.next_event_time += pd.Timedelta(days=1)
+    ## First Step
+    assert active_simulants(sim).empty
+    taken_step_size = take_step(sim)
+    assert taken_step_size == pd.Timedelta(days=1)
     for index in listener.event_indexes.values():
-        assert 0 not in index
-
-    # Check that everybody will step forward next step
-    assert len(active_simulants(sim)) == pop_size
+        assert index.empty
 
-    # Check that they are actually included in events
-    sim.step()
-    for index in listener.event_indexes.values():
-        assert 0 in index
-    # Revert change to 0
-    sim._population._population.step_size[0] /= 2
-    # Still step forward even with a non-integer step size
-    sim._population._population.step_size[7] /= 2
-    # Do a step just to update the next_event_time
-    sim.step()
-    # Check that next step, we will still update all
+    ## Second Timestep
     assert len(active_simulants(sim)) == pop_size
-    # Check that we actually include simulant of index 7 in events
-    sim.step()
+    taken_step_size = take_step(sim)
+    assert taken_step_size == pd.Timedelta(days=1)
     for index in listener.event_indexes.values():
-        assert 7 in index
-
-
-def test_step_pipeline(SimulationContext, base_config, components):
-    sim = SimulationContext(base_config, components)
-    sim.setup()
-    sim.initialize_simulants()
-
-    assert np.all(step_pipeline(sim) == step_column(sim))
-
-    sim.step()
-    assert np.all(step_pipeline(sim) == step_column(sim))
+        assert np.all(index == sim.get_population().index)
 
 
-@pytest.mark.parametrize("step_modifier", [0.5, 1, 2, 3.5, 5])
-def test_step_pipeline_with_modifier(SimulationContext, base_config, step_modifier):
-    sim = SimulationContext(base_config, [StepModifier("step_modifier", step_modifier)])
+@pytest.mark.parametrize(
+    "step_modifier_even,step_modifier_odd", [(0.5, 0.5), (1, 1), (2, 2), (4.5, 4)]
+)
+def test_skip_iterations(
+    SimulationContext, base_config, step_modifier_even, step_modifier_odd
+):
+    """Test that if everyone has some (non-minimum) step size, the global step adjusts to match"""
+    base_config["time"]["step_size"] = 1
+    listener = Listener("listener")
+    sim = SimulationContext(
+        base_config,
+        [StepModifier("step_modifier", step_modifier_even, step_modifier_odd), listener],
+    )
+    quantized_step = math.floor(max([step_modifier_even, 1]))
```

Comment:
This isn't the right name for this variable imo, since the quantized value is the minimum possible step size (i.e. 1). Better might be `expected_step_size`?

Reply:
This usage is common in physics--to "quantize" some continuous thing is to discretize it in terms of the quantum, so the "quantized" value is not the quantum itself, but the discretized representation of the thing in terms of the quantum. That said, I'm not really picky about the language and arguably it's better to avoid the somewhat inaccessible analogy to quantum mechanics in the first place.